project_task_analytic_propagation/models/account_analytic_line.py

```diff
@@ -7,18 +7,36 @@ class AccountAnalyticLine(models.Model):
     _inherit = "account.analytic.line"
 
     account_id = fields.Many2one(
-        compute="_compute_account_id",
-        store=True,
+        compute="_compute_analytic_accounts",
         readonly=False,
+        store=True,
     )
 
-    @api.depends("task_id.analytic_account_id")
-    def _compute_account_id(self):
-        """Change the analytic account non billed lines."""
+    def _get_analytic_plan_columns(self):
+        return [
+            f"task_id.{x}"
+            for x in self.task_id._get_plan_fnames()
+            if self._fields.get(x)
+        ]
+
+    @api.depends(lambda self: self._get_analytic_plan_columns())
+    def _compute_analytic_accounts(self):
+        plan_fnames = self._get_plan_fnames()
         timesheet_ids = self.filtered(
-            lambda t: not t.is_so_line_edited
-            and t._is_not_billed()
-            and t.task_id.analytic_account_id
+            lambda t: not t.is_so_line_edited and any(t[fname] for fname in plan_fnames)
         )
-        for timesheet in timesheet_ids:
-            timesheet.account_id = timesheet.task_id.analytic_account_id
+        for line in timesheet_ids:
+            vals_to_update = {}
+            for fname in plan_fnames:
+                if line._fields.get(fname) and line.task_id._fields.get(fname):
+                    vals_to_update[fname] = line.task_id[fname]
+            if vals_to_update:
+                line.write(vals_to_update)
```

Comment:
suggestion: no need to batch writes, no need to iterate twice, and [EAFP](https://docs.python.org/3/glossary.html#term-EAFP)


```suggestion
        for line in self:
            if t.is_so_line_edited:
                continue
            with suppress(KeyError):
                if line.task_id[fname]:
                    line[fname] = line.task_id[fname]
```

Remember to `from contextlib import suppress` at the start of the file.